nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp

```diff
@@ -49,39 +51,87 @@ GoalUpdater::GoalUpdater(
     rclcpp::SystemDefaultsQoS(),
     std::bind(&GoalUpdater::callback_updated_goal, this, _1),
     sub_option);
+  goals_sub_ = node_->create_subscription<geometry_msgs::msg::PoseStampedArray>(
+    goals_updater_topic,
+    rclcpp::SystemDefaultsQoS(),
+    std::bind(&GoalUpdater::callback_updated_goals, this, _1),
+    sub_option);
 }
 
 inline BT::NodeStatus GoalUpdater::tick()
 {
   geometry_msgs::msg::PoseStamped goal;
+  geometry_msgs::msg::PoseStampedArray goals;
 
   getInput("input_goal", goal);
+  getInput("input_goals", goals);
 
   // Spin multiple times due to rclcpp regression in Jazzy requiring a 'warm up' spin
   callback_group_executor_.spin_all(std::chrono::milliseconds(1));
   callback_group_executor_.spin_all(std::chrono::milliseconds(49));
 
-  if (last_goal_received_.header.stamp != rclcpp::Time(0)) {
-    auto last_goal_received_time = rclcpp::Time(last_goal_received_.header.stamp);
-    auto goal_time = rclcpp::Time(goal.header.stamp);
-    if (last_goal_received_time > goal_time) {
-      goal = last_goal_received_;
+  if (last_goal_received_set_) {
```

Comment:
I added these flags so that `last_goal_received_.header.stamp == rclcpp::Time(0)` is not valid  because it was not yet set